internal/server/data/pagination.go

```diff
@@ -16,10 +16,12 @@ func (p *Pagination) SetTotalCount(count int) {
 }
 
 func (p *Pagination) PaginateQuery(query *querybuilder.Query) {
-	if p.Page == 0 && p.Limit == 0 {
+	if p.Limit == 0 {
```

Comment:
should we be setting a default limit instead of returning?

Reply:
We do set a default limit in the API if the query comes in without a limit (`server.PaginationFromRequest`), but I don't think we want to set one here. Internal queries need to be allowed without a limit.

This is mostly just a safeguard, but I think we don't need it anymore. Since internal calls can use a `nil` to indicate they do not want pagination. I could remove this and rely on the defaults set in the API.